core/commonTest/src/OptionsTests.kt

```diff
@@ -143,4 +143,33 @@ class OptionsTests {
         assertEquals(ArgParser.ValueOrigin.REDEFINED, rendersOption.valueOrigin)
         assertEquals(ArgParser.ValueOrigin.REDEFINED, sourcesOption.valueOrigin)
     }
-}
+
+    object KeyValue: ArgType<Pair<String, String>>(true) {
+        override val description: kotlin.String
+            get() {
+                return "{ Value should have format <key>=<value> }"
+            }
+
+        override fun convert(value: kotlin.String, name: kotlin.String) =
+                if (!value.contains('=')) {
+                    throw ParsingException("Option $name should have value described with format <key>=<value>, but got $value")
+                } else {
+                    val (key, parameterValue) = value.split('=', limit = 2)
+                    key.trim() to parameterValue.trim()
+                }
+    }
+
+    @Test
+    fun testCustomOptionType() {
+        val argParser = ArgParser("testParser")
+        val useShortForm by argParser.option(ArgType.Boolean, shortName = "s", description = "Show short version of report").default(false)
+        val option by argParser.option(KeyValue, shortName = "op", description = "Additional options").required()
+        val output by argParser.option(ArgType.String, shortName = "o", description = "Output file")
+
+        argParser.parse(arrayOf("-o", "out.txt", "-op", "-Xrender=html"))
```

Comment:
I actually thought about a test, that would throw that exception and then you would catch it and assert it contained the expected string or so. Like it is done in ErrorsTest.kt.